lib/saveSecret.js

```diff
@@ -3,23 +3,83 @@
 const BbPromise = require('bluebird');
 const fs = require('fs');
 
-module.exports = {
-  saveSecret() {
-    if (!this.options.save) {
-      return BbPromise.resolve();
-    }
+const aws = require('aws-sdk')
 
-    let secrets = {};
+function file_data(location, cb) {
+  if (fs.existsSync(location)) {
+    fs.readFile(location, 'utf8', cb)
+  } else {
+    cb(null, "{}")
```

Comment:
If the file doesn't exist, I often view it as the empty object (unless there is a reason to need to work with missing configuration files).  I do similar things with the s3 bucket - 404 return.  Other errors, like permission denied, etc, will still propagate up.